pkg/manager/manager.go

```diff
@@ -120,6 +121,11 @@ type Options struct {
 	// Only use a custom NewCache if you know what you are doing.
 	NewCache cache.NewCacheFunc
 
+	// UserAgentID is the name of the manager to represent itself in the client requests.
+	//
+	// If set, the UserAgentID will be used to override the first part of rest.DefaultKubernetesUserAgent()
+	UserAgentID string
```

Comment:
What is the advantage of this over setting it in the rest config?

Reply:
> It re-uses DefaultKubernetesUserAgent, having the same structure and versioning as it was before, but with fixed os.Args[0] part.

I don't know what that means because I don't know how the default looks like. Why does it matter to re-use the same structure and versioning rather than setting the entire user-agent?

Reply:
Currently, DefaultKubernetesUserAgent is the useragent like: `manager/v1.0.1 (linux/amd64) kubernetes/12312afdb`, where we have os.Args[0] + version + runtime.Os/arch + git hash. 

So in the way of keep version and runtime the solution is to cut off the first part and replace it with correct name, rather than requiring operator to re-construct the whole UserAgent string with version and runtime lookups.

The better way to achieve it is to have a name param to client-go DefaultKubernetesUserAgent (or have the similar method) and set the UserAgent explicitely, but it is out of scope of controller-runtime.. But yes, I think I'd better do a PR to client-go, as it makes more sense there